src/profile-logic/process-profile.js

```diff
@@ -1163,25 +1168,28 @@ function sanitizeThreadPII(
   threadIndex: number,
   PIIToBeRemoved: RemoveProfileInformation
 ): Thread | null {
-  if (PIIToBeRemoved.shouldRemoveThreads.includes(threadIndex)) {
+  if (PIIToBeRemoved.shouldRemoveThreads.has(threadIndex)) {
     // If this is a hidden thread, remove the thread immediately.
     // This will not remove the thread entry from the `threads` array right now
     // and just replace it with a `null` value. We filter out the null values
     // inside `serializeProfile` function.
     return null;
   }
+  // Deep copying the thread since we are gonna mutate it and we don't want to alter
+  // current thread.
   // We need to update the stringTable. It's not possible with UniqueStringArray.
   const stringArray = thread.stringTable.serializeToArray();
+  let markerTable = cloneRawMarkerTable(thread.markers);
```

Comment:
As we talked this on Slack, I'm going to keep that name for now since there are also other functions like `cloneFrameTable` and `cloneFuncTable` in the same file with the same approach. I'm gonna open a PR to rename them all later.